tokio/src/runtime/io/scheduled_io.rs

```diff
@@ -510,14 +511,22 @@ cfg_io_readiness! {
                         drop(waiters);
                     }
                     State::Done => {
-                        let tick = TICK.unpack(scheduled_io.readiness.load(Acquire)) as u8;
-
                         // Safety: State::Done means it is no longer shared
                         let w = unsafe { &mut *waiter.get() };
 
+                        // Note: the returned tick might be newer than the event
+                        // which notified our waker. This is ok because the future
+                        // still didn't return `Poll::Ready`.
+                        let curr = scheduled_io.readiness.load(Acquire);
+                        let tick = TICK.unpack(curr) as u8;
+
+                        // Add more readiness which might have appeared in the meantime.
+                        let curr_ready = Ready::from_usize(READINESS.unpack(curr));
+                        let ready = w.ready | (curr_ready.intersection(w.interest));
```

Comment:
If a readiness bit is not included in `curr_ready` then something else already cleared it. Including `w.ready` here would result in additional false positives & unnecessary I/O ops.

Based on discord, it sounds like you want to avoid empty readiness events, but that seems better to me than false positives. The caller would loop and wait again.